extensions/wikia/RobotsTxt/tests/RobotTest.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+use Wikia\RobotsTxt\Robot;
+
+class RobotTest extends WikiaBaseTest {
+
+	public function setUp() {
+		global $IP;
+		$this->setupFile = "$IP/extensions/wikia/RobotsTxt/RobotsTxt.setup.php";
+		parent::setUp();
+	}
+
+	/**
+	 * Test Wikia\RobotsTxt\Robot API
+	 *
+	 * @covers       Robot::allowPaths
+	 * @covers       Robot::disallowPaths
+	 * @covers       Robot::setUserAgent
+	 * @covers       Robot::getContent
+	 * @dataProvider dataProviderClassApi
+	 *
+	 * @param string[]|null $allowPaths argument passed to $robot->allowPaths (null for don't call it)
+	 * @param string[]|null $disallowPaths argument passed to $robot->disallowPaths (null for don't call it)
+	 * @param string[]|null $userAgent argument passed to $robot->setUserAgent (null for don't call it)
+	 * @param string[] $expectedContents expected $robots->getContent()
+	 */
+	public function testClassApi( $allowPaths, $disallowPaths, $userAgent, $expectedContents ) {
+		$robot = new Robot( '*' );
+		
+		if ( !is_null( $userAgent ) ) {
+			$robot->setUserAgent( $userAgent );
+		}
+		if ( !is_null( $allowPaths ) ) {
+			$robot->allowPaths( $allowPaths );
+		}
+		if ( !is_null( $disallowPaths ) ) {
+			$robot->disallowPaths( $disallowPaths );
+		}
+		$this->assertEquals( $expectedContents, $robot->getContent() );
+	}
+
+	/**
+	 * Test Wikia\RobotsTxt\Robot API
+	 *
+	 * @covers       Robot::allowPaths
+	 * @covers       Robot::disallowPaths
+	 * @covers       Robot::setUserAgent
+	 * @covers       Robot::getContent
+	 * @dataProvider dataProviderClassApi
+	 *
+	 * @param string[]|null $allowPaths argument passed to $robot->allowPaths (null for don't call it)
+	 * @param string[]|null $disallowPaths argument passed to $robot->disallowPaths (null for don't call it)
+	 * @param string[]|null $userAgent argument passed to $robot->setUserAgent (null for don't call it)
+	 * @param string[] $expectedContents expected $robots->getContent()
+	 */
+	public function testClassApiTwistedOrder( $allowPaths, $disallowPaths, $userAgent, $expectedContents ) {
+		$robot = new Robot( '*' );
+		
+		if ( !is_null( $userAgent ) ) {
+			$robot->setUserAgent( $userAgent );
+		}
+		if ( !is_null( $allowPaths ) ) {
+			$robot->allowPaths( $allowPaths );
+		}
+		if ( !is_null( $disallowPaths ) ) {
+			$robot->disallowPaths( $disallowPaths );
+		}
+		$this->assertEquals( $expectedContents, $robot->getContent() );
+	}
+
+	/**
+	 * Test robot blocking
+	 *
+	 * @covers       Robot::block
+	 * @covers       Robot::getContent
+	 */
+	public function testRobotBlocking() {
+		$robot = new Robot( '*' );
+
+		$robot->block();
+
+		$this->assertEquals(
+			[
+				'User-agent: *',
+				'Disallow: /',
+				''
+			],
+			$robot->getContent()
+		);
+	}
+
+	public function dataProviderClassApi() {
+		return [
```

Comment:
You can use PHP generators (`yield`) in data providers which makes them more readable.